### PR TITLE
[PROGRAMMES-5837] Is today method refactor

### PR DESCRIPTION
**Pull request by tomwinnington**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5837

When iterating 1000 times, I found it saved 2 milliseconds.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php [hunk 1]
```diff
@@ -32,6 +33,11 @@ public function isGmt(): bool
         return $this->options['user_timezone'] == 'GMT';
     }
 
+    public function isToday(): bool
+    {
+        return ApplicationTime::getTime()->format('Y-m-d') == $this->getDateTime()->toDateString();
```
Comment (votemike) [inline src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php]: Should be able to still use `->toDateString()` on the ApplicationTime

Comment (benolot) [reply src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php]: There's no method like that in the Application Time class, and as far as I understood, format was the best way to format a `DateTimeImmutable` object?

File: src/Ds2013/Presenters/Utilities/DateList/date_list.html.twig [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <ol class="nav nav--banner date-list {{ date_list.getOption('css_classes') }}">
+
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Utilities/DateList/date_list.html.twig]: ARGH WHITESPACE ARGH ARGHGHGHGH!!!!!111111ELEVEN

File: src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php (hunk 1 above)
Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php]: In Practice, ApplicationTime will return Chronos, not DateTimeImmutable, so what mike's saying is right, if counterintuitve

File: src/Ds2013/Presenters/Utilities/DateList/date_list.html.twig (hunk 2 above)
Comment (votemike) [reply src/Ds2013/Presenters/Utilities/DateList/date_list.html.twig]: ARGHGHGHGH!!!!!sin(90)cos(0)

File: src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php [hunk 3]
```diff
@@ -32,6 +33,11 @@ public function isGmt(): bool
         return $this->options['user_timezone'] == 'GMT';
     }
 
+    public function isToday(): bool
```
Comment (votemike) [inline src/Ds2013/Presenters/Utilities/DateList/DayDateListItemPresenter.php]: I think a small doc block explaining why we aren't just using `$this->getDateTime()->isToday()` would be useful.

**Pull request merged**